File: ASM.Server/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ASM.Server.Models
{
	public class Order
	{
		public int Id { get; set; }

		public string ShippingAddress { get; set; }
		public OrderStatus Status { get; set; }
		public decimal TotalAmount { get; set; }
		public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

		public string UserId { get; set; }

		[ForeignKey(nameof(UserId))]
		public AppUser? User { get; set; }

		public ICollection<OrderDetail>? OrderDetails { get; set; }
	}

	public enum OrderStatus
	{
		Pending = 1, // làm món
		Shipping = 2, // đang giao hàng
		Delivered = 3, // đã giao
	}
}

File: asm.client/src/types/order.ts
export interface Order {
  id: number;
  shippingAddress: string;
  status: OrderStatus;
  totalAmount: number;
  createdAt: string;
  updatedAt?: string;
  userId: string;
  discountAmount: string;
  paymentMethod: PaymentMethod;
  orderDetails?: OrderDetail[] | null;
}

export interface OrderDetail {
  id: number;
  quantity: number;
  unitPrice: number;
  productId?: number | null;
}

export enum PaymentMethod {
  COD = 1,
  Momo = 2,
}

export enum OrderStatus {
  Pending = 1, // làm món
  Shipping = 2, // đang giao hàng
  Delivered = 3, // đã giao
}
